live-flowchart/app.js

```diff
@@ -0,0 +1,73 @@
+// import Logger from 'logger';
+// import SWUtil from 'sw-util';
+
+function App() {
+  var _self = this;
+
+  Logger.log('App()');
+
+  // instatiate a new Service Worker Utility Class
+  this.swUtil = new SWUtil();
+
+  // register click events
+  document.querySelector('#reloadapp').addEventListener('click', function() {
+    window.location.reload();
+  });
+
+  // checking whether service workers are supported
+  if (this.swUtil.areServiceWorkersSupported()) {
+    document.querySelector('#swinstall').addEventListener('click', function() {
+      Logger.log('\n-------\n');
+      _self.enableCoolFeatures();
+    });
+
+    document.querySelector('#swuninstall').addEventListener('click', function() {
+      Logger.log('\n-------\n');
+      _self.disableCoolFeatures();
+    });
+
+    // checking whether a service worker is in control
+    if (this.swUtil.isServiceWorkerControllingThisApp()) {
+      Logger.info('App code run as expected');
+
+      this.disableServiceWorkerRegistration();
+    } else {
+      this.enableServiceWorkerRegistration();
+    }
+  } else {
+    Logger.error('Service workers are not supported by this browser');
+  }
+}
+
+App.prototype = {
+
+  enableCoolFeatures: function enableCoolFeatures() {
+    Logger.log('\nApp.enableCoolFeatures()');
+
+    Logger.log('Configuring service worker');
+
+    this.swUtil.registerServiceWorker(
```

Comment:
This might benefit from using promises to go hand in hand with SW's design paradigms.